scipy/fftpack/basic.py

```diff
@@ -20,6 +20,12 @@
 atexit.register(_fftpack.destroy_rfft_cache)
 del atexit
 
+try:
+    # no half-precision routines in fftpack.  run float16 in single precision
+    float16_code = numpy.dtype('float16').char
```

Comment:
half-float exists since numpy 1.6.0. You can assume it is always available.
You don't need to declare `float16_code`, just write `dtype == np.half` below.